### PR TITLE
vimPlugins.minimap-vim: init at 2021-01-04

### PR DESCRIPTION
**Pull request by teto**
adds a minimap in vim, quite cool IMO

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tnias** [comment]
If I manually put code-minimap into the PATH, it works just fine.

If I just add it to vim as a plugin, I get the following error. I assume the substitution did not work as expected.

```
/nix/store/...-vimplugin-minimap-vim-2021-01-04/share/vim-plugins/minimap-vim/bin/minimap_generator.sh: line 19: code-minimap: command not found^@
```

To reproduce the error: Open vim and run `:Minimap`.

**Comment by teto** [comment]
thanks for trying, I had added code-minimap in neovim's PATH as a first approach which is why I had not caught the issue.
I've patched that binary too, seems to work here.

**Comment by tnias** [comment]
The new version works. :+1:

If I am in NORMAL mode on the code side and scroll very fast, it starts lagging. Fast scrolling in INSERT mode does not cause lagging. But I assume this is an issue with the plugin not the packaging.

File: pkgs/misc/vim-plugins/overrides.nix [hunk 1]
```diff
@@ -14,7 +14,7 @@
 , buildVimPluginFrom2Nix
 , nodePackages
 , dasht
-, sqlite
+, sqlite, code-minimap
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:17]:
```suggestion
, sqlite
, code-minimap
```

File: pkgs/misc/vim-plugins/overrides.nix [hunk 2]
```diff
@@ -243,6 +243,15 @@ self: super: {
     dependencies = with super; [ webapi-vim ];
   });
 
+  minimap-vim = super.minimap-vim.overrideAttrs(old: {
+    preFixup = ''
+      substituteInPlace $out/share/vim-plugins/minimap-vim/plugin/minimap.vim --replace "code-minimap" \
+      "${code-minimap}/bin/code-minimap"
+      substituteInPlace $out/share/vim-plugins/minimap-vim/bin/minimap_generator.sh --replace "code-minimap" \
+      "${code-minimap}/bin/code-minimap"
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:251]:
```suggestion
      substituteInPlace $out/share/vim-plugins/minimap-vim/plugin/minimap.vim \
        --replace "code-minimap" "${code-minimap}/bin/code-minimap"
      substituteInPlace $out/share/vim-plugins/minimap-vim/bin/minimap_generator.sh \
        --replace "code-minimap" "${code-minimap}/bin/code-minimap"
```

We might want to use wrapProgram.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111150` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>spacevim</li>
</ul>
</details>

**Comment by teto** [comment]
thanks

**Pull request merged**